<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>EditDialog</class>
 <widget class="QDialog" name="EditDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>618</width>
    <height>382</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Edit database cell</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QPushButton" name="buttomImport">
       <property name="toolTip">
        <string>Import text</string>
       </property>
       <property name="whatsThis">
        <string>Opens a file dialog used to import text to this database cell.</string>
       </property>
       <property name="text">
        <string>&amp;Import</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="buttonExport">
       <property name="toolTip">
        <string>Export text</string>
       </property>
       <property name="whatsThis">
        <string>Opens a file dialog used to export the contents of this database cell to a text file.</string>
       </property>
       <property name="text">
        <string>&amp;Export</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QComboBox" name="comboEditor">
       <item>
        <property name="text">
         <string>Text</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Binary</string>
        </property>
       </item>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="buttonClear">
       <property name="toolTip">
        <string>Clear cell data</string>
       </property>
       <property name="whatsThis">
        <string>Erases the contents of the cell</string>
       </property>
       <property name="text">
        <string>&amp;Clear</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QStackedWidget" name="editorStack">
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="verticalLayout_4">
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <item>
        <widget class="QLabel" name="labelBinayWarning">
         <property name="text">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600; color:#c00000;&quot;&gt;Warning: Editing binary content in text mode may result in corrupted data!&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QTextEdit" name="editorText">
         <property name="font">
          <font>
           <family>Monospace</family>
          </font>
         </property>
         <property name="whatsThis">
          <string>This area displays information about the data present in this database cell</string>
         </property>
         <property name="acceptRichText">
          <bool>false</bool>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="editorBinary"/>
     <widget class="QScrollArea" name="editorImageScrollArea">
      <property name="widgetResizable">
       <bool>true</bool>
      </property>
      <widget class="QWidget" name="scrollAreaWidgetContents">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>0</y>
         <width>94</width>
         <height>24</height>
        </rect>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <widget class="QLabel" name="editorImage">
          <property name="text">
           <string/>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </widget>
   </item>
   <item>
    <widget class="QLabel" name="labelType">
     <property name="text">
      <string>Type of data currently in cell</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QLabel" name="labelSize">
     <property name="text">
      <string>Size of data currently in table</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <tabstops>
  <tabstop>buttomImport</tabstop>
  <tabstop>buttonExport</tabstop>
  <tabstop>comboEditor</tabstop>
  <tabstop>buttonClear</tabstop>
  <tabstop>buttonBox</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>EditDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>262</x>
     <y>370</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>EditDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>326</x>
     <y>350</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttomImport</sender>
   <signal>clicked()</signal>
   <receiver>EditDialog</receiver>
   <slot>importData()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>41</x>
     <y>16</y>
    </hint>
    <hint type="destinationlabel">
     <x>60</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonExport</sender>
   <signal>clicked()</signal>
   <receiver>EditDialog</receiver>
   <slot>exportData()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>100</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>162</x>
     <y>17</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonClear</sender>
   <signal>clicked()</signal>
   <receiver>EditDialog</receiver>
   <slot>clearData()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>321</x>
     <y>13</y>
    </hint>
    <hint type="destinationlabel">
     <x>290</x>
     <y>12</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>editorText</sender>
   <signal>textChanged()</signal>
   <receiver>EditDialog</receiver>
   <slot>editTextChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>234</x>
     <y>218</y>
    </hint>
    <hint type="destinationlabel">
     <x>362</x>
     <y>309</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>comboEditor</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>editorStack</receiver>
   <slot>setCurrentIndex(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>265</x>
     <y>16</y>
    </hint>
    <hint type="destinationlabel">
     <x>185</x>
     <y>169</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>editorStack</sender>
   <signal>currentChanged(int)</signal>
   <receiver>comboEditor</receiver>
   <slot>setCurrentIndex(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>185</x>
     <y>169</y>
    </hint>
    <hint type="destinationlabel">
     <x>267</x>
     <y>16</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>importData()</slot>
  <slot>exportData()</slot>
  <slot>clearData()</slot>
  <slot>editTextChanged()</slot>
  <slot>checkDataType()</slot>
 </slots>
</ui>
